composer_triggering_dags_with_functions_and_pubsub/main.tf

```diff
@@ -0,0 +1,226 @@
+#                       #
+#########################
+# Comment this section to use an existing service account
+resource "google_service_account" "composer_env_sa" {
+  project      = data.google_project.project.project_id
+  account_id   = "composer-worker-sa"
+  display_name = "Test Service Account for Composer Environment deployment "
+}
+
+resource "google_project_service_identity" "composer_sa" {
+  provider = google-beta
+  project  = data.google_project.project.project_id
+  service  = "composer.googleapis.com"
+}
+
+resource "google_project_iam_member" "composer_worker" {
+  project = data.google_project.project.project_id
+  role    = "roles/composer.worker"
+  member  = "serviceAccount:${google_service_account.composer_env_sa.email}"
+}
+
+resource "google_service_account_iam_member" "custom_service_account" {
+  provider           = google-beta
+  service_account_id = google_service_account.composer_env_sa.id
+  role               = "roles/composer.ServiceAgentV2Ext"
+  member             = "serviceAccount:${google_project_service_identity.composer_sa.email}"
+}
+
+########################
+#                      #
+# Creates PubSub topic #
+#                      #
+########################
+resource "google_pubsub_topic" "trigger" {
+  project                    = data.google_project.project.project_id
+  name                       = "dag-topic-trigger"
+  message_retention_duration = "86600s"
+}
+
+##########################
+#                        #
+# Creates Cloud Function #
+#                        #
+##########################
+resource "google_cloudfunctions_function" "pubsub_function" {
+  project = data.google_project.project.project_id
+  name    = "pubsub-publisher"
+  runtime = "python310"
+  region  = "us-central1"
+
+  available_memory_mb   = 128
+  source_archive_bucket = google_storage_bucket.cloud_function_bucket.name
+  source_archive_object = "pubsub_function.zip"
+  timeout               = 60
+  entry_point           = "pubsub_publisher"
+  service_account_email = "${data.google_project.project.number}-compute@developer.gserviceaccount.com"
+  trigger_http          = true
+
+}
+
+##################################
+#                                #
+# Create Cloud Storage resources #
+#                                #
+##################################
+
+resource "google_storage_bucket" "cloud_function_bucket" {
+  project                     = data.google_project.project.project_id
+  name                        = "${data.google_project.project.project_id}-cloud-function-source-code"
+  location                    = "US"
+  force_destroy               = true
+  uniform_bucket_level_access = true
+}
+
+resource "google_storage_bucket_object" "cloud_function_source" {
+  name   = "pubsub_function.zip"
+  bucket = google_storage_bucket.cloud_function_bucket.name
+  # Uncomment to upload a zip containing the cloudfunction source code in zip format
+  #source = "/path/to/cloudfunction/source.zip"
```

Comment:
I think commenting out the source attribute isn't helping. When I try to run the sample without the zip file present in the directory,  I get errors like this:

```
Error: Error deleting contents of object pubsub_function.zip: Delete "https://storage.googleapis.com/storage/v1/b/ci-tf-samples-1-ade9-cloud-function-source-code/o/pubsub_function.zip?alt=json&prettyPrint=false": dial tcp [2607:f8b0:400e:c09::80]:443: connect: cannot assign requested address
```